src/figuresClasses.ts

```diff
@@ -1,11 +1,83 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  readonly shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+type NumberValue = number;
+
+export class Triangle implements Figure {
+  readonly shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: NumberValue = 0,
+    public b: NumberValue = 0,
+    public c: NumberValue = 0,
+  ) {
```

Comment:
[CODE KNOWLEDGE] - When declaring a constructor parameter with an access modifier, TypeScript implicitly creates a property on the class with the same name as the parameter and assigns the parameter value to that property. You don't need to manually assign them inside the constructor body. You can remove the assignments and leave the constructor body empty.